pkg/infra/lifecyclehooks.go

```diff
@@ -84,27 +84,29 @@ var _ = Describe("Lifecycle Hooks should", framework.LabelMachines, func() {
 
 			return false, nil
 		}, framework.WaitLong, pollingInterval).Should(BeTrue(), "Pod did not start running on machine")
+
+		DeferCleanup(func() {
+			By("Deleting the machineset")
+			cascadeDelete := metav1.DeletePropagationForeground
+			Expect(client.Delete(context.Background(), machineSet, &runtimeclient.DeleteOptions{
+				PropagationPolicy: &cascadeDelete,
+			})).To(Succeed(), "MachineSet should be able to be deleted")
+
+			By("Waiting for the MachineSet to be deleted...")
+			framework.WaitForMachineSetsDeleted(ctx, client, machineSet)
+
+			By("Deleting workload job")
+			Expect(client.Delete(context.Background(), workload, &runtimeclient.DeleteOptions{
+				PropagationPolicy: &cascadeDelete,
+			})).To(Succeed(), "Workload job should be able to be deleted")
+		})
```

Comment:
Nit: It would be preferable to put the cleanup next to the thing that creates it, I would have split this into several cleanups so that, if something breaks mid setup, we tear down the bits we have already created

The flow should be:
* Create MachineSet
* Defer remove MachineSet
* Wait for MachineSet
* Create workload Job
* Defer remove workload Job
* Wait for workload job running